apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx

```diff
@@ -362,6 +375,7 @@ export const DeviceActionDefaultRendering = <R, H extends States, P>({
           swapDefaultTrack,
           amountExpectedTo: amountExpectedTo.toString() ?? undefined,
           estimatedFees: estimatedFees?.toString() ?? undefined,
+          getCurrencySettings: useCurrencySettingsSelector,
```

Comment:
we probably should rename the `getCurrencySettings` function with a `useCurrencySettings` naming? (in order for linter to correctly apply the use* rules)

Reply:
Can't do it because it throw a lint error about usage Hook

React Hook "useCurrencySettings" is called in function "renderSwapDeviceConfirmation" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".eslint[react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html)

Reply:
ok, then i think it's a problem, renderSwapDeviceConfirmation must not use hooks, since they are indeed not a component.

we probably need to pass in the whole Settings store, and lense from it the necessary information.

This is what i had to do for the name too: https://github.com/LedgerHQ/ledger-live/blob/398e3e09e55db9369bbd861eaf8b384e77220704/apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx#L367